### DISCUSSION TITLE
Data-graph conversion operation

### DISCUSSION THREAD

**Discussion by AslantheAslan** [snap-stanford/ogb#367] [MAG240M-LSC]
After I downloaded the dataset, I wanted to get graph and init_labels such as follows:
```
    dataset = MAG240MDataset(root = 'data/')
    g, init_labels = dataset[0]
```
However, that makes me end up with the error `'MAG240MDataset' object is not subscriptable`. I have done the same thing with a similar dataset called ogbn-mag but I never faced with such an issue. I would be more than happy if I can learn how I can get g and init_labels.

Edit: I got that data types of "ogbn-mag" and "ogb-mag240m" are a bit different so I have converted the data object to a heterogeneous graph.  Actually, I am still converting due to the huge amount of nodes and edges but I suppose it will work.

**Comment by weihua916** [comment]
Hi! The interfaces of OGB datasets and LSC datasets are different. See the tutorial [here](https://ogb.stanford.edu/docs/lsc/mag240m/#dataset_object) for how to use the LSC datasets.

**Reply by AslantheAslan** [reply]
Thank you very much for your response. Right after my question, I found out that I needed to convert it to a graph, a DGL graph in particular. I have been working on this dataset and am still having issues fitting it in a single memory even without node features. It is regardless of the title of this discussion, but that would be better than anything if someone from the OGB team have published an article or shared some scripts on how we can utilize the MAG240M-LSC dataset without consuming an excessive amount of memory. Right now, I spend all my time finding a way to fit the data into CPU memory, rather than working on my model. However, I know that this is one of the biggest challenges of this competition.

Wish you all the best.
